test/Encumber.t.sol

```diff
@@ -0,0 +1,322 @@
+
+        // alice encumbers the remainder of her balance to charlie
+        cometWrapper.encumber(charlie, 150e18);
+
+        vm.stopPrank();
+
+        assertEq(cometWrapper.balanceOf(alice), 200e18);
+        assertEq(cometWrapper.availableBalanceOf(alice), 0);
+        assertEq(cometWrapper.encumberedBalanceOf(alice), 200e18);
+        assertEq(cometWrapper.encumbrances(alice, bob), 50e18);
+        assertEq(cometWrapper.encumbrances(alice, charlie), 150e18);
+        assertEq(cometWrapper.allowance(alice, bob), type(uint256).max);
+
+        // bob calls transfers from alice, attempting to transfer his encumbered
+        // tokens and also transfer tokens encumbered to charlie
+        vm.prank(bob);
+        vm.expectRevert(CometWrapper.InsufficientAvailableBalance.selector);
+        cometWrapper.transferFrom(alice, bob, 100e18);
+    }
+
+    function test_transferFrom_revertsIfInsufficientAllowance() public {
+        deal(address(cometWrapper), alice, 100e18);
+
+        vm.startPrank(alice);
+
+        // alice encumbers some of her balance to bob
+        cometWrapper.encumber(bob, 10e18);
+
+        // she also grants him an approval
+        cometWrapper.approve(bob, 20e18);
+
+        vm.stopPrank();
+
+        assertEq(cometWrapper.balanceOf(alice), 100e18);
+        assertEq(cometWrapper.availableBalanceOf(alice), 90e18);
+        assertEq(cometWrapper.encumberedBalanceOf(alice), 10e18);
+        assertEq(cometWrapper.encumbrances(alice, bob), 10e18);
+        assertEq(cometWrapper.allowance(alice, bob), 20e18);
+        assertEq(cometWrapper.balanceOf(charlie), 0);
+
+        // bob tries to transfer more than his encumbered and allowed balances
+        vm.prank(bob);
+        vm.expectRevert(CometWrapper.InsufficientAllowance.selector);
+        cometWrapper.transferFrom(alice, charlie, 40e18);
+    }
+
+    function test_encumberFrom_revertsOnInsufficientAllowance() public {
+        deal(address(cometWrapper), alice, 100e18);
+
+        // alice grants bob an approval
+        vm.prank(alice);
+        cometWrapper.approve(bob, 50e18);
+
+        // but bob tries to encumber more than his allowance
+        vm.prank(bob);
+        vm.expectRevert(CometWrapper.InsufficientAllowance.selector);
+        cometWrapper.encumberFrom(alice, charlie, 60e18);
+    }
+
+    function test_encumberFrom() public {
+        deal(address(cometWrapper), alice, 100e18);
+
+        // alice grants bob an approval
+        vm.prank(alice);
+        cometWrapper.approve(bob, 100e18);
+
+        assertEq(cometWrapper.balanceOf(alice), 100e18);
+        assertEq(cometWrapper.availableBalanceOf(alice), 100e18);
+        assertEq(cometWrapper.encumberedBalanceOf(alice), 0e18);
+        assertEq(cometWrapper.encumbrances(alice, bob), 0e18);
+        assertEq(cometWrapper.allowance(alice, bob), 100e18);
+        assertEq(cometWrapper.balanceOf(charlie), 0);
+
+        // but bob tries to encumber more than his allowance
```

Comment:
does this comment apply to this test?

bob has an allowance of 100 and he's encumbering 60

Reply:
Good catch. Took this from the encumber factory repo so the test there probably has same incorrect comment. I'll make a small fix there as well.